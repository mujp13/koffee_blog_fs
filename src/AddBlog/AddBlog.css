.add-blog-form {
  padding: 20px;
  width: 100%;
  height: 100%;
  font-size: 20px;
}

label {
  float: left;
}

select {
  width: 10%;
  height: 25px;
}

textarea {
  width: 30%;
  height: 50px;
  font-size: 20px;
}

.submit-button {
  width: 100px;
  height: 30px;
}
