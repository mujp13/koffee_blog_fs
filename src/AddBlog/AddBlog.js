import React, { Component } from 'react';
import KoffeeBlogContext from '../KoffeeBlogContext/KoffeeBlogContext';
import './AddBlog.css';

export default class AddBlog extends Component {
  static contextType = KoffeeBlogContext;

  state = {
    error: null,
  };

  handleSubmit = (e) => {
    e.preventDefault();
    // get the form fields from the event
    const { typeid, title, imageurl, content, rating, location } = e.target;
    const contents = {
      typeid: typeid.value,
      title: title.value,
      imageurl: imageurl.value,
      content: content.value,
      rating: rating.value,
      location: location.value,
    };

    this.setState({ error: null });

    fetch('https://friendly-donair-37840.herokuapp.com/api/contents', {
      method: 'POST',
      body: JSON.stringify(contents),
      headers: {
        'content-type': 'application/json',
      },
    })
      .then((res) => {
        if (!res.ok) {
          // get the error message from the response,
          return res.json().then((error) => {
            // then throw it
            throw error;
          });
        }
        return res.json();
      })
      .then((data) => {
        console.log(data);
        this.context.addBlog(data);
        this.props.history.push('/');
      })
      .catch((error) => {
        this.setState({ error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <>
        <h2>Add your own blog post here!</h2>
        <form className="add-blog-form" onSubmit={this.handleSubmit}>
          <div className="add-blog" role="alert">
            {error && <p>{error.messsage}</p>}
          </div>
          <p>Type</p>
          <select name="typeid" id="typeid">
            <option value="Bean">Bean</option>
            <option value="Iced Coffee">Iced Coffee</option>
            <option value="Brewing">Brewing</option>
            <option value="Drinks">Drinks</option>
          </select>
          <div>
            {' '}
            <p>Title</p>
            <textarea name="title" id="title" placeholder="(i.e. Delicious Coffe)" required></textarea>
          </div>
          <div>
            <p>Image URL(i.e. "...nameofphoto.jpg")</p>
            <textarea name="imageurl" id="imageurl" placeholder="(i.e. www.webiste.com/photo.jpg)"></textarea>
          </div>
          <div>
            {' '}
            <p>Description</p>
            <textarea name="content" id="content" placeholder="Put your description" required></textarea>
          </div>
          <div>
            {' '}
            <p>Rating(1~5)</p>
            <textarea
              name="rating"
              id="rating"
              placeholder="(1: worst, 2: bad, 3: so-so, 4: great, 5: best)"
            ></textarea>
          </div>
          <div>
            {' '}
            <p>Location</p>
            <textarea name="location" id="location" placeholder="(i.e. CA, TX, WA)"></textarea>
          </div>
          <br></br>
          <input className="submit-button" type="submit" value="Submit"></input>
        </form>
      </>
    );
  }
}
