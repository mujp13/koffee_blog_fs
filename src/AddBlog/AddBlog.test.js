import React from 'react';
import ReactDOM from 'react-dom';
import AddBlog from './AddBlog';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

//smoke
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <AddBlog />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

//snapshot
it('renders correctly', () => {
  const tree = renderer.create(<AddBlog />).toJSON();
  expect(tree).toMatchSnapshot();
});
