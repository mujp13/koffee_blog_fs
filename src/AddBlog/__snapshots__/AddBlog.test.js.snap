// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2>
    Add your own blog post here!
  </h2>,
  <form
    className="add-blog-form"
    onSubmit={[Function]}
  >
    <div
      className="add-blog"
      role="alert"
    />
    <p>
      Type
    </p>
    <select
      id="typeid"
      name="typeid"
    >
      <option
        value="Bean"
      >
        Bean
      </option>
      <option
        value="Iced Coffee"
      >
        Iced Coffee
      </option>
      <option
        value="Brewing"
      >
        Brewing
      </option>
      <option
        value="Drinks"
      >
        Drinks
      </option>
    </select>
    <div>
       
      <p>
        Title
      </p>
      <textarea
        id="title"
        name="title"
        placeholder="(i.e. Delicious Coffe)"
        required={true}
      />
    </div>
    <div>
      <p>
        Image URL(i.e. "...nameofphoto.jpg")
      </p>
      <textarea
        id="imageurl"
        name="imageurl"
        placeholder="(i.e. www.webiste.com/photo.jpg)"
      />
    </div>
    <div>
       
      <p>
        Description
      </p>
      <textarea
        id="content"
        name="content"
        placeholder="Put your description"
        required={true}
      />
    </div>
    <div>
       
      <p>
        Rating(1~5)
      </p>
      <textarea
        id="rating"
        name="rating"
        placeholder="(1: worst, 2: bad, 3: so-so, 4: great, 5: best)"
      />
    </div>
    <div>
       
      <p>
        Location
      </p>
      <textarea
        id="location"
        name="location"
        placeholder="(i.e. CA, TX, WA)"
      />
    </div>
    <br />
    <input
      className="submit-button"
      type="submit"
      value="Submit"
    />
  </form>,
]
`;
